esp/src/src-react/components/XrefDirectories.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from "react";
+import { CommandBar, ContextualMenuItemType, ICommandBarItemProps } from "@fluentui/react";
+import { useConst } from "@fluentui/react-hooks";
+import { scopedLogger } from "@hpcc-js/util";
+import { HolyGrail } from "../layouts/HolyGrail";
+import * as WsDFUXref from "src/WsDFUXref";
+import * as Observable from "dojo/store/Observable";
+import { Memory } from "src/Memory";
+import { useGrid } from "../hooks/grid";
+import { ShortVerticalDivider } from "./Common";
+import nlsHPCC from "src/nlsHPCC";
+
+const logger = scopedLogger("src-react/components/XrefDirectories.tsx");
+
+interface XrefDirectoriesProps {
+    name: string;
+}
+
+export const XrefDirectories: React.FunctionComponent<XrefDirectoriesProps> = ({
+    name
+}) => {
+
+    //  Grid ---
+    const store = useConst(new Observable(new Memory("name")));
+    const [Grid, _selection, refreshTable, copyButtons] = useGrid({
+        store,
+        query: {},
+        sort: [{ attribute: "name", "descending": false }],
+        columns: {
+            name: { width: 100, label: nlsHPCC.Name },
+            num: { width: 30, label: nlsHPCC.Files },
+            size: { width: 30, label: nlsHPCC.TotalSize },
+            maxIP: { width: 30, label: nlsHPCC.MaxNode },
+            maxSize: { width: 30, label: nlsHPCC.MaxSize },
+            minIP: { width: 30, label: nlsHPCC.MinNode },
+            minSize: { width: 30, label: nlsHPCC.MinSize },
+            positiveSkew: {
+                width: 30,
+                label: nlsHPCC.SkewPositive,
+                renderCell: (object, value, node, options) => {
+                    if (value === undefined) {
+                        return "";
+                    }
+                    node.innerText = value;
+                }
+            },
+            negativeSkew: {
+                width: 30,
+                label: nlsHPCC.SkewNegative,
+                renderCell: (object, value, node, options) => {
+                    if (value === undefined) {
+                        return "";
+                    }
+                    node.innerText = value;
+                }
+            }
+        },
+        filename: "xrefsDirectories"
+    });
+
+    //  Command Bar  ---
+    const buttons = React.useMemo((): ICommandBarItemProps[] => [
+        {
+            key: "refresh", text: nlsHPCC.Refresh, iconProps: { iconName: "Refresh" },
+            onClick: () => refreshTable()
```

Comment:
This won't refresh the data (a somewhat common issue with local Memory based stores and our React implementation...)
See:  https://track.hpccsystems.com/browse/HPCC-26594